src/tempo.py

```diff
@@ -167,8 +171,14 @@ def is_ready(self):
             return False
         return out == "ready"
 
-    def _build_receivers_config(self, receivers: Sequence[ReceiverProtocol]):  # noqa: C901
-        receivers_set = set(receivers)  # convert to set for faster lookup
+    def _build_receivers_config(self, remote_receivers: Sequence[ReceiverProtocol]):  # noqa: C901
+        # remote_receivers: the receivers we have to enable because the requirers we're related to
+        # intend to use them
+        # self._enabled_receivers: receivers we have to enable because *this charm* will use them.
+        receivers_set = set(remote_receivers)
+        receivers_set.update(self._enabled_receivers)
+        if not receivers_set:
+            logger.warning("No receivers set. Tempo will be up but not functional.")
```

Comment:
IMO if there is/was something like charm metrics that could be a reason for an alert.

Reply:
yeah on the operator side I'm not sure if it would be reasonable to put the charm in blocked
fact is, until you relate it, tempo has no reason to even start if not to monitor itself (via charm_tracing)

and in practice, thanks to charm_tracing, there will always be at least `otlp_http` in there so this should never happen.